src/main/java/io/cryostat/recordings/LongRunningRequestGenerator.java

```diff
@@ -0,0 +1,267 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.recordings;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+import io.cryostat.ConfigProperties;
+import io.cryostat.reports.ReportsService;
+import io.cryostat.ws.MessagingServer;
+import io.cryostat.ws.Notification;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.quarkus.vertx.ConsumeEvent;
+import io.vertx.mutiny.core.eventbus.EventBus;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+
+@ApplicationScoped
+public class LongRunningRequestGenerator {
+
+    public static final String ARCHIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ArchiveRequest";
+    public static final String GRAFANA_ARCHIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.GrafanaArchiveUploadRequest";
+    public static final String GRAFANA_ACTIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.GrafanaActiveUploadRequest";
+    public static final String ARCHIVE_REPORT_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ArchiveReportRequest";
+    public static final String ACTIVE_REPORT_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ActiveReportRequest";
+    private static final String ARCHIVE_RECORDING_SUCCESS = "ArchiveRecordingSuccess";
+    private static final String ARCHIVE_RECORDING_FAIL = "ArchiveRecordingFailed";
+    private static final String GRAFANA_UPLOAD_SUCCESS = "GrafanaUploadSuccess";
+    private static final String GRAFANA_UPLOAD_FAIL = "GrafanaUploadFailed";
+    private static final String REPORT_SUCCESS = "ReportSuccess";
+    private static final String REPORT_FAILURE = "ReportFailure";
+    private final ExecutorService executor;
+
+    @Inject Logger logger;
+    @Inject private EventBus bus;
+    @Inject private RecordingHelper recordingHelper;
+    @Inject private ReportsService reportsService;
+
+    @ConfigProperty(name = ConfigProperties.CONNECTIONS_FAILED_TIMEOUT)
+    Duration timeout;
+
+    public LongRunningRequestGenerator(ExecutorService executor) {
+        this.executor = executor;
```

Comment:
Do we still need this separate executor for archiving? It seems like the Grafana upload and report generation tasks work well with just using a generic worker thread with `@ConsumeEvent(blocking = true)`.

Reply:
nope, this can be cleaned up now. Initially it was there since I took inspiration from how the InterruptibleReportGenerator worked (including the name :)) but the blocking annotation is a better way of handling this. Cleaned up now